emma-examples/emma-examples-flink/src/test/scala/org/emmalanguage/examples/ml/clustering/FlinkKMeansIntegrationSpec.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2014 TU Berlin (emma@dima.tu-berlin.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.emmalanguage
+package examples.ml.clustering
+
+import KMeans.Solution
+import api.Meta.Projections._
+import api._
+import examples.ml.model._
+
+import breeze.linalg.Vector
+import org.apache.flink.api.scala.ExecutionEnvironment
+
+class FlinkKMeansIntegrationSpec extends BaseKMeansIntegrationSpec {
+
+  override def kMeans(k: Int, epsilon: Double, iterations: Int, input: String): Set[Solution[Long]] =
+    /*emma.onFlink*/ {
```

Comment:
There is another problem with `KMeans` besides the `Meta` context bound on case classes. `TranslateToDataflows` doesn't introduce `fetch()` calls for method parameters captured in higher order context. This can be observed with the variable `centroids`. We discussed a possible fix with @ggevay.